File: moedict-android/app/src/main/java/com/yichiuan/moedict/BuilderModule.java
package com.yichiuan.moedict;

import com.yichiuan.moedict.ui.main.MainActivity;
import com.yichiuan.moedict.ui.search.SearchActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;


@Module
public abstract class BuilderModule {
    @ContributesAndroidInjector
    abstract MainActivity bindMainActivity();

    @ContributesAndroidInjector
    abstract SearchActivity bindSearchActivity();
}

File: moedict-android/app/src/main/java/com/yichiuan/moedict/BuilderModule.kt
package com.yichiuan.moedict

import com.yichiuan.moedict.ui.main.MainActivity
import com.yichiuan.moedict.ui.search.SearchActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector


@Module
abstract class BuilderModule {
    @ContributesAndroidInjector
    internal abstract fun bindMainActivity(): MainActivity

    @ContributesAndroidInjector
    internal abstract fun bindSearchActivity(): SearchActivity
}
